diff --git a/sub/lavc_conv.c b/sub/lavc_conv.c
index 532e91d508..655299f4ad 100644
--- a/sub/lavc_conv.c
+++ b/sub/lavc_conv.c
@@ -94,8 +94,10 @@ struct lavc_conv *lavc_conv_create(struct mp_log *log,
     av_dict_set(&opts, "sub_text_format", "ass", 0);
 #endif
     av_dict_set(&opts, "flags2", "+ass_ro_flush_noop", 0);
-    if (strcmp(priv->codec, "eia_608") == 0)
+    if (strcmp(priv->codec, "eia_608") == 0) {
         av_dict_set(&opts, "real_time", "1", 0);
+        av_dict_set(&opts, "data_field", "0", 0);
+    }
     if (avcodec_open2(avctx, codec, &opts) < 0)
         goto error;
     av_dict_free(&opts);
